src/services/BudgetService.ts

```diff
@@ -130,4 +150,40 @@ export class BudgetService {
       )
     }
   }
+
+  static getProposalsBudgetingMinAndMaxDates(proposals: Pick<ProposalAttributes, 'start_at'>[]): {
+    minDate?: Date
+    maxDate?: Date
+  } {
+    const sorted = proposals
+      .filter((p) => p.start_at && p.start_at >= BUDGETING_START_DATE)
+      .sort((p1, p2) => p1.start_at.getTime() - p2.start_at.getTime())
+    return { minDate: sorted[0]?.start_at, maxDate: sorted[sorted.length - 1]?.start_at }
+  }
+
+  static async getBudgets(proposals: Pick<ProposalAttributes, 'start_at'>[]): Promise<CurrentBudget[]> {
+    const budgetsForProposals: CurrentBudget[] = []
+    const { minDate, maxDate } = this.getProposalsBudgetingMinAndMaxDates(proposals)
+    if (!minDate) return budgetsForProposals
+    const oldestBudget = await QuarterBudgetModel.getBudget(minDate)
+    if (oldestBudget === null) {
+      ErrorService.report(`Could not find budget for ${minDate}`)
+      return budgetsForProposals
+    }
+    budgetsForProposals.push(oldestBudget)
+    if (maxDate && minDate !== maxDate) {
+      const newestBudget = await QuarterBudgetModel.getBudget(maxDate)
+      if (newestBudget !== null) {
+        const index = budgetsForProposals.findIndex((budget) => budget.id === newestBudget.id)
```

Comment:
Is this for checking if newestBudget is the same is the oldestBudget?
What if you do something like...
```
const newestBudget = await QuarterBudgetModel.getBudget(maxDate)
if(newestBudget !== null && newestBudget.id !== oldestBudget.id) {
  budgetsForProposals.push(newestBudget)
}
```